### PR TITLE
Add prebuilt component: StakeERC721Button

### PR DESCRIPTION
**Pull request by kien-ngo**
## PR-Codex overview
The focus of this PR is to add a new `erc20Value` property to various generated files in the modular extensions.

### Detailed summary
- Added `erc20Value` property to multiple generated files for modular extensions.

> The following files were skipped due to too many changes: `packages/thirdweb/src/extensions/modular/__generated__/BatchMetadataERC1155/write/uploadMetadata.ts`, `packages/thirdweb/src/extensions/modular/__generated__/RoyaltyERC721/write/setDefaultRoyaltyInfo.ts`, `packages/thirdweb/src/extensions/modular/__generated__/RoyaltyERC1155/write/setDefaultRoyaltyInfo.ts`, `packages/thirdweb/src/extensions/modular/__generated__/RoyaltyERC721/write/setRoyaltyInfoForToken.ts`, `packages/thirdweb/src/extensions/modular/__generated__/RoyaltyERC1155/write/setRoyaltyInfoForToken.ts`, `packages/thirdweb/src/extensions/modular/__generated__/DelayedRevealBatchMetadataERC721/write/reveal.ts`, `packages/thirdweb/src/extensions/modular/__generated__/MintableERC20/write/completeOwnershipHandover.ts`, `packages/thirdweb/src/extensions/modular/__generated__/ClaimableERC1155/write/setClaimConditionByTokenId.ts`, `packages/thirdweb/src/extensions/modular/__generated__/OpenEditionMetadataERC721/write/setSharedMetadata.ts`, `packages/thirdweb/src/extensions/modular/__generated__/OpenEditionMetadataERC1155/write/setSharedMetadata.ts`, `packages/thirdweb/src/extensions/modular/__generated__/DelayedRevealBatchMetadataERC721/write/uploadMetadata.ts`, `packages/thirdweb/scripts/generate/abis/erc721/IStaking721.json`, `packages/thirdweb/src/exports/react.ts`, `packages/thirdweb/src/react/web/ui/prebuilt/thirdweb/CreateDirectListingButton/CreateDirectListingButton.test.tsx`, `.changeset/cold-toys-type.md`, `packages/thirdweb/src/react/web/ui/prebuilt/thirdweb/StakeERC721Button/StakeERC721Button.test.tsx`, `packages/thirdweb/src/extensions/erc721/__generated__/IStaking721/read/stakingToken.ts`, `packages/thirdweb/src/react/web/ui/prebuilt/thirdweb/StakeERC721Button/index.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/thirdweb/src/extensions/erc721/__generated__/IStaking721/read/stakingToken.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ * ```ts
+ * import { stakingToken } from "thirdweb/extensions/erc721";
+ *
+ * const result = await stakingToken({
+ *  contract,
+ * });
+ *
+ * ```
+ */
+export async function stakingToken(options: BaseTransactionOptions) {
```
Comment (gregfromstl) [inline packages/thirdweb/src/extensions/erc721/__generated__/IStaking721/read/stakingToken.ts:66]: Changeset for this

File: packages/thirdweb/src/extensions/modular/__generated__/BatchMetadataERC1155/write/uploadMetadata.ts [hunk 2]
```diff
@@ -153,5 +153,6 @@ export function uploadMetadata(
       (await asyncOptions()).overrides?.maxPriorityFeePerGas,
     nonce: async () => (await asyncOptions()).overrides?.nonce,
     extraGas: async () => (await asyncOptions()).overrides?.extraGas,
+    erc20Value: async () => (await asyncOptions()).overrides?.erc20Value,
```
Comment (gregfromstl) [inline packages/thirdweb/src/extensions/modular/__generated__/BatchMetadataERC1155/write/uploadMetadata.ts:156]: What are these for? Also were these added via the generation script or manual? I can't find the diff for the generation script

**Review by gregfromstl — CHANGES_REQUESTED** [review]
Mainly just curious what `erc20Value` is for and if there's a way we can avoid modifying the extension options

**Comment by kien-ngo** [comment]
closing this for now, need to rethink the design

**Pull request closed without merging**